### PR TITLE
Update find-bugs to generate a report

### PR DESCRIPTION
**Pull request by smunilla**
find-bugs has value as it lets us define exactly which bugs will be
attached to an advisory based on declaritive filters. This update will
allow users to see more useful information about those bugs other than
just the IDs.

### CODE REVIEW

File: elliottlib/bzutil.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def search_for_bugs(bz_data, status, search_filter='default', filter_out_securit
     if verbose:
         click.echo(query_url)
     
-    return _perform_query(bzapi, query_url)
+    return _perform_query(bzapi, query_url, include_fields=['id', 'status', 'summary', 'creation_time', 'cf_pm_score', 'component'])
```
Comment (tbielawa) [inline elliottlib/bzutil.py]: Does this slow the query down? I'm thinking of @ scale when we're finding like 500+ bugs to attach. If it does add a penalty we might only want to add those additional fields if report is requested.

Obvious followup: "Why not yaml?"

File: elliott [hunk 2]
```diff
@@ -351,8 +351,12 @@ advisory.
...
+@click.option("--report",
+              required=False,
+              is_flag=True,
+              help="Output a detailed report of the found bugs")
 @pass_runtime
-def find_bugs(runtime, advisory, default_advisory_type, auto, status, id, flag):
+def find_bugs(runtime, advisory, default_advisory_type, auto, status, id, flag, report):
     """Find Red Hat Bugzilla bugs or add them to ADVISORY. Bugs can be
 "swept" into the advisory either automatically (--auto), or by
 manually specifying one or more bugs using the --id option. Mixing
```
Comment (tbielawa) [inline elliott]: Maybe mention what fields are included in the report?

File: elliottlib/bzutil.py (hunk 1 above)
Comment (smunilla) [reply elliottlib/bzutil.py]: I actually don't know if that affects the speed or not. The cli tool seems to pull all the fields in by default. But, the fact that this is a feature makes me think it probably has some effect, and it is easy enough to make this a conditional.

**Comment by tbielawa** [comment]
Hm. What am I doing wrong @smunilla?


```
[~/rhat/cd/elliott] 7:53:32  (master)
$ git co add-report
Branch 'add-report' set up to track remote branch 'add-report' from 'smunilla'.
Switched to a new branch 'add-report'
[~/rhat/cd/elliott] 7:53:35  (add-report)
$ time ./elliott --group=openshift-4.1 find-bugs --auto --report
/home/tbielawa/.config/elliott/settings.yaml
2019-04-17 07:54:04,456 INFO Cloning config data from git@github.com:openshift/ocp-build-data.git
2019-04-17 07:54:07,109 INFO Using branch from group.yml: rhaos-4.1-rhel-7
Searching for bugs with target release(s): 4.1.0, 4.1.z 
Found 608 bugs: 1655414, 1686282, 1670618, 1664559, ...
...
...
1692281, 1531938, 1674346, 1631517, 1668242, 1656670, 1677236, 1679791, 1645463, 1675381
ID       COMPONENT                 STATUS       SCORE   AGE        SUMMARY                                                     
Traceback (most recent call last):
  File "./elliott", line 906, in <module>
    main()
  File "./elliott", line 895, in main
    cli(obj={})
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/tbielawa/.local/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "./elliott", line 447, in find_bugs
    bug.cf_pm_score,
  File "/usr/lib/python2.7/site-packages/bugzilla/bug.py", line 115, in __getattr__
    raise AttributeError(msg)
AttributeError: Bug object has no attribute 'cf_pm_score'.
If 'cf_pm_score' is a bugzilla attribute, it may not have been cached when the bug was fetched. You may want to adjust your include_fields for getbug/query.
./elliott --group=openshift-4.1 find-bugs --auto --report  0.79s user 0.09s system 6% cpu 13.876 total
```

Weeeeird.

I added a little debug....

```diff
$ git diff         
diff --git a/elliott b/elliott
index 3f19328..0bd8ddb 100755
--- a/elliott
+++ b/elliott
@@ -441,6 +441,8 @@ advisory with the --add option.
         for bug in bug_ids:
             created_date = datetime.datetime.strptime(str(bug.creation_time), '%Y%m%dT%H:%M:%S')
             days_ago = (datetime.datetime.today() - created_date).days
+            green_print("Inspecting bug {}".format(bug.id))
+            print(dir(bug))
             click.echo("{:<8d} {:<25s} {:<12s} {:<7s} {:<3d} days   {:60s} ".format(bug.id,
                                                                                     bug.component,
                                                                                     bug.status,
```



```
ID       COMPONENT                 STATUS       SCORE   AGE        SUMMARY                                                     
Inspecting bug 1655414
['__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattr__', '__getattribute__', '__getstate__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_e
x__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__unicode__', '__weakref__', '_bug_fields', '_update_dict', 'addcc', 'addcomment', 'autorefresh', 'bugzilla', 'close', 'component', 'components', 'creation_time', 'deletecc', 'get_attachment_ids', 'get_attachments', 'get_flag_status', 'get_flag_type', 'get_flags', 'get_history_raw', 'getcomments', 'id', 'refresh', 'reload', 'setassignee', 'setstatus', 'status', 'summary', 'updateflags', 'weburl']
Traceback (most recent call last):
...
...
...
...
AttributeError: Bug object has no attribute 'cf_pm_score'.
If 'cf_pm_score' is a bugzilla attribute, it may not have been cached when the bug was fetched. You may want to adjust your include_fields for getbug/query.
./elliott --group=openshift-4.1 find-bugs --auto --report  0.82s user 0.08s system 12% cpu 7.201 total
```

Seems that the bug it was inspecting actually doesn't have that field https://bugzilla.redhat.com/show_bug.cgi?id=1655414

**Comment by tbielawa** [comment]
It appears that my bugzilla client token had expired. After refreshing it with `bugzilla login` I stopped getting the errors above

File: elliottlib/bzutil.py (hunk 1 above)
Comment (tbielawa) [reply elliottlib/bzutil.py]: I just did some quick checks and the speed was the same with and without `report` :+1:

**Comment by sosiouxme** [comment]
@smunilla i missed this somehow and looks like it's fallen by the wayside. Just in the last couple weeks I've been wishing for this. Any chance you could rebase this and make a gitlab MR?

**Comment by thiagoalessio** [comment]
Closed in favor of https://github.com/openshift/elliott/pull/58

**Pull request closed without merging**